epic.js

```diff
@@ -149,33 +177,37 @@ const createEpics = ({ actionTypes, actionCreators, service }) => {
   const isEndAction = either(isCompleteAction, isErrorAction)
   const requestHandlers = createRequestHandlers(actionCreators)
   const mapRequestHandlers = mapObjIndexed((requestHandler, method) => {
-    return (action$, store, deps) => {
+    return (action$, state$, deps) => {
       assertFeathersDep(deps)
 
       const { feathers } = deps
 
       const requester = createRequester({ method, feathers, service })
 
-      return action$.ofType(actionTypes[method])
-        .mergeMap(action => {
+      return action$.pipe(
+        ofType(actionTypes[method]),
+        mergeMap(action => {
           const args = action.payload
           const { cid } = action.meta
 
           const response$ = requester(args)
-          const requestAction$ = requestHandler(response$, { cid, args, service, store })
-
-          const cidAction$ = action$.filter(isCid(cid))
-          const completeAction$ = cidAction$.filter(isCompleteAction)
-          const errorAction$ = cidAction$.filter(isErrorAction)
-          const cancelAction$ = completeAction$.merge(errorAction$)
-
-          return Rx.Observable.of(actionCreators.start(cid, { service, method, args }))
-            .concat(requestAction$)
-            .concat(Rx.Observable.of(actionCreators.complete(cid)))
-            .catch(err => Rx.Observable.of(actionCreators.error(cid, err)))
-            .takeUntil(cancelAction$)
-            .filter(endOnce(isEndAction))
+          const requestAction$ = requestHandler(response$, { cid, args, service, state$ })
+
+          const cidAction$ = action$.pipe(rxFilter(isCid(cid)))
+          const completeAction$ = cidAction$.pipe(rxFilter(isCompleteAction))
+          const errorAction$ = cidAction$.pipe(rxFilter(isErrorAction))
+          const cancelAction$ = completeAction$.pipe(rxMerge(errorAction$))
+
+          return of(actionCreators.start(cid, { service, method, args })).pipe(
+            concat(requestAction$),
+            concat(of(actionCreators.complete(cid))),
+            catchError(err => of(actionCreators.error(cid, err))),
+            takeWhile(endOnce(isEndAction))
+            // takeUntil(cancelAction$),
```

Comment:
how do you cancel then?

Reply:
have just gotten around to looking at this now :| hmm, correct me if i'm wrong 🙏 but isn't a cancel action just a 'complete' action? from what i see in `feathers-action-react`? the function `isEndAction` there is an `either` of complete or error actions, so my understanding is that `takeWhile` will take until `endOnce` is false, which happens when an 'endAction' is passed in. Though now I look at `endOnce` more closely, it looks like it will wait one more value of the stream until doing so... hmmm

as far as tests go, test 'find with cancel' should be testing this behaviour i think? https://github.com/root-systems/feathers-action/blob/49409f435bb0a47e78f38e418e4ec6f26477af7c/test/epic.js#L54